pkg/sources/redisstreams/redisstream.go

```diff
@@ -228,6 +229,93 @@ func newRedisClient(sourceSpec *dfv1.RedisStreamsSource) (*redisclient.RedisClie
 	return redisclient.NewRedisClient(opts), nil
 }
 
+// incoming message should have multiple key/value pairs: produce a message with no Key, with value set to JSON serialization of the pairs
+func produceUnkeyedJSONMsg(inMsg redis.XMessage) (*isb.ReadMessage, error) {
+	var readOffset = inMsg.ID
+
+	// JSON serialize our key/value pairs
+	// need to first convert map[string]interface to map[string]string
+	kvMap := make(map[string]string)
+	for k, v := range inMsg.Values {
+		strValue, castOk := v.(string)
```

Comment:
Why can't we convert the original `map[string]interface{}` to json directly? I believe all the value types are okay to marshal.

Reply:
You're right - I'd need to do a `[]byte assertion` and then cast to string. So, apparently the underlying type must be a string in the tests I've done so far. Hmmm...